### ISSUE TITLE
How to do error handling?

### ISSUE THREAD

**Issue by munjalpatel** [jgaskins/grpc#5]
First of all, this is simply great! Awesome work!!

I am trying to figure out how to properly return errors on exceptions.
Any suggestions?

**Comment by jgaskins** [comment]
Good point. I don't think there's anything in here yet for it for returning any [response code](https://developers.google.com/maps-booking/reference/grpc-api/status_codes) other than 0 (OK). I think this is the reason the Ruby implementation requires you to [accept the `call` object](https://grpc.io/docs/languages/ruby/basics/#implementing-routeguide) in your method, so you can change the status.

I think it'd be a great idea to at least rescue exceptions and return an internal error if this shard doesn't already do that — it's been a while since I looked at that part of the code. 🙂

**Comment by munjalpatel** [comment]
Yeah, I think two things are missing:
1. Error / Exception Management
2. Authentication / Metadata support

**Comment by bendangelo** [comment]
Those are two pretty big issues. Are there any plans on implementing them?

**Comment by jgaskins** [comment]
I'm open to suggestions

**Comment by bendangelo** [comment]
I've opened a PR to add status codes. For auth I'm not sure where to start. https://github.com/jgaskins/grpc/pull/13

**Comment by bendangelo** [comment]
Also I've noticed, the standard "list" command to see available services is not available. This would be a nice addition.
